components/director/internal/domain/api/service.go

```diff
@@ -201,13 +228,24 @@ func (s *service) GetFetchRequest(ctx context.Context, apiDefID string) (*model.
 	return fetchRequest, nil
 }
 
-func (s *service) createFetchRequest(ctx context.Context, tenant string, in model.FetchRequestInput, parentObjectID string) (*string, error) {
+func (s *service) createFetchRequest(ctx context.Context, tenant string, in model.FetchRequestInput, parentObjectID string) (*model.FetchRequest, error) {
 	id := s.uidService.Generate()
 	fr := in.ToFetchRequest(s.timestampGen(), id, tenant, model.APIFetchRequestReference, parentObjectID)
 	err := s.fetchRequestRepo.Create(ctx, fr)
 	if err != nil {
 		return nil, errors.Wrapf(err, "while creating FetchRequest for %s with ID %s", model.APIFetchRequestReference, parentObjectID)
 	}
 
-	return &id, nil
+	return fr, nil
+}
+
+func (s *service) handleFetchRequest(ctx context.Context, api *model.APIDefinition, fr *model.FetchRequest) {
```

Comment:
This logic seems to be repeating in every service using fetch requests, maybe it could be moved to fetch request service?